Preview menus

vincetiu8 opened the pull request:
This adds both tower and enemy preview menus into the main menu so the user can preview what different towers and enemies do. Note they cannot edit any attributes, as this would be problematic. Former 'Tower Preview', 'Enemy Preview' and 'Level Preview' menus have been renamed to 'Tower Edit', 'Enemy Edit' and 'Level Edit' respectively. Note the tower and enemy preview code is in devtools, so don't delete that file!

vincetiu8 committed:
Add a tower preview menu

This adds a tower preview menu, to remain in the final release, where the player can view information about the tower but is unable to edit any attributes.

vincetiu8 committed:
Add an enemy preview menu

This adds an enemy preview menu where the player can preview enemies, similar to the tower preview menu.

Graphcalibur reviewed (changes requested):
LGTM, just need to resolve the merge conflicts

vincetiu8 committed:
Merge branch 'dev' into preview-menus

vincetiu8 committed:
Fix merge conflicts

vincetiu8 committed:
Fix enemy bug

This fixes a bug where if we changed the enemy type the tower would continue shooting at the position of its former enemy and wouldn't lock onto any of the new enemies.

Graphcalibur reviewed (approved):
LGTM!
